### PR TITLE
Improve test coverage for raw decoder

### PR DESCRIPTION
**Pull request by charlesjmorgan**
~~Missed this method in #4183, fixes error when getting value from `RawValueProvider` more than once. Need to "reset" the position each time a get method is called.~~

_Thought_ I missed `getSlice` but I did not. Instead I've just improved the test to cover every combination of presto/java types.

### CODE REVIEW

File: presto-record-decoder/src/main/java/io/prestosql/decoder/raw/RawColumnDecoder.java [hunk 1]
```diff
@@ -289,6 +289,7 @@ private void checkEnoughBytes()
         @Override
         public Slice getSlice()
         {
+            value.position(start);
```
Comment (losipiuk) [inline presto-record-decoder/src/main/java/io/prestosql/decoder/raw/RawColumnDecoder.java:292]: good catch. Can we have test for that?

Comment (charlesjmorgan) [reply presto-record-decoder/src/main/java/io/prestosql/decoder/raw/RawColumnDecoder.java:292]: yes I'll add one/update the old one

**Comment by charlesjmorgan** [comment]
added columns for each different combination of data types

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationSmokeTest.java [hunk 2]
```diff
@@ -78,13 +92,39 @@ protected QueryRunner createQueryRunner()
     @Test
     public void testColumnReferencedTwice()
     {
-        ByteBuffer buf = ByteBuffer.allocate(8);
-        buf.putLong(0, 1);
+        ByteBuffer buf = ByteBuffer.allocate(48);
```
Comment (losipiuk) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationSmokeTest.java:95]: Use larger values which span all the bytes for needed for given data type, so we are sure we are reading everything.
If you use `1` as test value for 8-byte char we can be reading just a single byte and still not-fail assertion. Sth like `0x1234567890abcdef` will give better coverage.

**Comment by losipiuk** [comment]
Merged, thanks!

**Pull request merged**